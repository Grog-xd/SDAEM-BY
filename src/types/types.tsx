//svg

export interface Svg{
    width?: string,
    height?: string,
    color?: string,
    style?: string,
}

// products

export interface ProductItemProps{
    params?: paramsProps,
    likeActive?: boolean,
    handler?: () => void,
    product?: ProductProps,
    id?:number,
    like?:boolean
}

export interface ProductProps{
    img?: any,
    type?:string,
    cost?:string|number,
    rooms?: number,
    title?: string,
    district?: string,
    city?:string,
    location?:string,
    people?: string | number,
    metro?: string,
    body?: string,
    id?: number,
    obj?: string,
    profile?: object,
    size?: string,
    length?: number
}

//params

export interface paramsProps{
    type?:string,
    id?:number
}

//news

export interface newProps{
    img?: string,
    title?: string,
    body?: string,
    date?: string,
    id?: number,
    bigImg?: string,
    description?: string,
    length?: number,
    params?: paramsProps
}

//option

export interface optionSelectProps{
    id:number,
    value: string | number,
    bol: boolean,
}

// login redux

export interface loginRedux {
    login:{
        isAuth?: boolean,
        profilesArr?:profilesArr,
        profile?:profilesArr
    }
}

export interface profilesArr{
    name?: string,
    avatar?: any,
    phone?: string,
    city?: string,
    email?: string,
    password?: string,
    viber?: string,
    whatsUpp?: string,
    length?: number
}

// main redux

export interface mainRedux {
    main:{
        tabs?: optionSelectProps[],
        cityOption?: optionSelectProps[],
        cityCurrentValue?: string,
        roomsOption?:optionSelectProps[],
        roomsCurrentValue?:string,
        metroOption?:optionSelectProps[],
        metroCurrentValue?:string,
        districtOption?:optionSelectProps[],
        districtCurrentValue?:string,
        productsLimit?:number,
        currentPage?:number,
        sortedOptions?: optionSelectProps[],
        sortedValue?: string,
        bedSumOptions?: optionSelectProps[],
        bedSumCurrentValue?:string,
        checkboxsMoreOption?: optionSelectProps[],
        minCost?: string | number,
        maxCost?: string | number,
        productsLen?: number,
        sortedProducts?:ProductProps,
        products?:ProductProps,

    }
}

// news redux

export interface newsRedux{
    news:{
        sortedPosts?: newProps,
        limitNewsItems?: number,
        currentPage?: number,
        contactsInformation?:contactsInformationProps,
        posts?:newProps
    }
}

export interface contactsInformationProps{
    location?: string,
    tel?: string | number,
    email?: string,
    timeWork?: string
}
