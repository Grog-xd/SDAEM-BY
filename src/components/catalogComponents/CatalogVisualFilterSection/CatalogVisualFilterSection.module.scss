@import "../../../style/mixins";

.visualFilter{
  margin-top: 21.5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.sortedSelect{
  width: 232px;
  background: #FFFFFF;
  box-shadow: 0 5px 20px rgba(0, 96, 206, 0.1);
  border-radius: 18px;
  p{
    font-weight: 600;
  }
}

.btnsBlock{
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.setLimitBtns{
  display: flex;
  align-items: center;
}


.btnActive{
  margin-right: 21.5px;
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  cursor: pointer;
  background: #FFFFFF;
  box-shadow: 0 5px 20px rgba(0, 96, 206, 0.1);
  border-radius: 18px;
  padding: 10px 15px;
  @include text(14px, 600, 17px, #1E2123);
  svg{
    margin-right: 10px;
  }
}


.btn{
  margin-right: 21.5px;
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  cursor: pointer;
  filter: drop-shadow(0px 5px 20px rgba(0, 96, 206, 0.1));
  border-radius: 18px;
  padding: 10px 15px;
  @include text(14px, 500, 17px, #1E2123);
  &:hover{
    background: #F8F8F8;
  }
  svg{
    margin-right: 10px;
  }
}

.map{
  background: #FFFFFF;
  margin-left: 43.5px;
  box-shadow: 0 5px 20px rgba(0, 96, 206, 0.1);
  border-radius: 18px;
  padding: 10px 15px;
  @include text(14px, 600, 17px, #1E2123);
  &:hover{
    color: #664EF9;
  }
  svg{
    margin-right: 10px;
  }
}
