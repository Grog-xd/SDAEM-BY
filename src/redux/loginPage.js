import {createSlice} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, setBookMarkActive, registerUser} = loginPage.actions

