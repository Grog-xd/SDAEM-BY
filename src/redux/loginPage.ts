import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'
import {profilesArr} from '../types/types';

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action:PayloadAction<profilesArr>){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action:PayloadAction<any>){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, registerUser} = loginPage.actions

