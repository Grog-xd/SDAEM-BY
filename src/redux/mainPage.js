import {createSlice} from '@reduxjs/toolkit';

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:'',
        maxCost:'',


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action){
            state.sortedValue = action.payload
        },

        setLimit(state, action){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = ''
            state.maxCost = ''
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = +state.maxCost + 100
            }
            if(state.minCost === ''){
                state.minCost=0
            }
            if(state.maxCost === ''){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
