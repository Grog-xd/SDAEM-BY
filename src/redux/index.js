import {combineReducers} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import mainPage from "./mainPage";
import newsPage from "./newsPage";
import loginPage from "./loginPage";

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage
})

export const store = configureStore({
    reducer: rootReducer
})
