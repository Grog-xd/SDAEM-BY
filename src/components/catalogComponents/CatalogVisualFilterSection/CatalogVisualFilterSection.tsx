import {FC} from 'react';
import {Link, useParams} from 'react-router-dom';
import {useDispatch, useSelector} from 'react-redux';

import {mainFilter, setLimit, setSortedValue} from '../../../redux/mainPage';
import MySelect from '../../UI/mySelect/mySelect.tsx';
import SvgSort from '../../svg/SvgSort.tsx';
import SvgBurger from '../../svg/SvgBurger.tsx';
import SvgTiles from '../../svg/SvgTiles.tsx';
import SvgLocation from '../../svg/SvgLocation.tsx';

import {mainRedux} from '../../../types/types';

import classes from './CatalogVisualFilterSection.module.scss';

const CatalogVisualFilterSection:FC = () => {
    const dispatch = useDispatch()
    const params = useParams()

    const {productsLimit, sortedOptions, sortedValue} = useSelector((state:mainRedux)=> state.main)

    function limitHandler(limit:number){
        dispatch(setLimit(limit))
        dispatch(mainFilter(params.type))
    }

    function sortHandler(value:string){
        dispatch(setSortedValue(value))
        dispatch(mainFilter(params.type))
    }

    const defaultLimit = 6

    return (
        <section className={classes.visualFilter}>
            <MySelect options={sortedOptions} style={classes.sortedSelect} value={sortedValue ? sortedValue: 'По умолчанию'} handler={(e)=>sortHandler(e) }>
                {
                    sortedValue
                        ?
                        <SvgSort width={'15'} height={'16'} color={'#664EF9'}/>
                        :
                        <SvgSort width={'15'} height={'16'} color={'#7A7F86'}/>
                }

            </MySelect>
            <div className={classes.btnsBlock}>
                {
                    productsLimit === defaultLimit
                        ?
                        <div className={classes.setLimitBtns}>
                            <button className={classes.btn} onClick={()=>limitHandler(3)}>
                                <SvgBurger width={'15'} height={'15'} color={'#7A7F86'}/>
                                Список
                            </button>
                            <button className={classes.btnActive}>
                                <SvgTiles width={'15'} height={'14'} color={'#664EF9'}/>
                                Плитки
                            </button>
                        </div>
                        :
                        <div className={classes.setLimitBtns}>
                            <button className={classes.btnActive} >
                                <SvgBurger width={'15'} height={'15'} color={'#664EF9'}/>
                                Список
                            </button>
                            <button className={classes.btn} onClick={()=>limitHandler(6)}>
                                <SvgTiles width={'15'} height={'14'} color={'#7A7F86'}/>
                                Плитки
                            </button>
                        </div>
                }
                <Link to={'/404'} className={classes.map}>
                    <SvgLocation width={'11'} height={'14'} color={'#664EF9'}/>
                    Показать на карте
                </Link>
            </div>
        </section>
    );
};

export default CatalogVisualFilterSection;
