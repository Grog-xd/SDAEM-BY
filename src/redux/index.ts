import {combineReducers} from 'redux';
import {configureStore} from '@reduxjs/toolkit';
import {useDispatch} from 'react-redux';

import mainPage from './mainPage';
import newsPage from './newsPage';
import loginPage from './loginPage';

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage,
})

export const store = configureStore({
    reducer: rootReducer,
})


export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
