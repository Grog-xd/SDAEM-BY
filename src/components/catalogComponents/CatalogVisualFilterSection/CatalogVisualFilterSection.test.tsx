import {render, screen} from '@testing-library/react';
import {Provider, useSelector} from 'react-redux';

import {MemoryRouter} from 'react-router-dom';

import {store} from '../../../redux';

import {mainRedux} from '../../../types/types';

import CatalogVisualFilterSection from './CatalogVisualFilterSection';

describe('catalog visual filter section tests', ()=>{
    test('snapshot test', ()=>{
        render(
            <Provider store={store} >
                <MemoryRouter >
                    <CatalogVisualFilterSection />
                </MemoryRouter>
            </Provider>,
        )
        const section = screen.getByTestId('catalog-visual-filter-section')
        expect(section).toMatchSnapshot();
    })
})
