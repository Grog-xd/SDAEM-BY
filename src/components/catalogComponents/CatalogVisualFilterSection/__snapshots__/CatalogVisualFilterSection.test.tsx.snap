// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalog visual filter section tests snapshot test 1`] = `
<section
  class="visualFilter"
  data-testid="catalog-visual-filter-section"
>
  <div
    class="select sortedSelect"
  >
    <button
      class="changeBlock"
      type="button"
    >
      <div>
        <svg
          fill="none"
          height="16"
          viewBox="0 0 15 16"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_2831_2828)"
          >
            <g
              opacity="0.6"
            >
              <path
                d="M7.32328 6.74139H11.4807C11.5823 6.74139 11.6647 6.65952 11.6647 6.55807V4.37278C11.6647 4.27148 11.5826 4.18945 11.4807 4.18945H7.32328C7.22199 4.18945 7.13965 4.27133 7.13965 4.37278V6.55822C7.13965 6.65952 7.22183 6.74139 7.32328 6.74139Z"
                fill="#7A7F86"
              />
              <path
                d="M7.32328 3.07178H9.67264C9.77393 3.07178 9.85627 2.9896 9.85627 2.88815V0.703167C9.85627 0.601561 9.77424 0.519531 9.67264 0.519531H7.32328C7.22199 0.519531 7.13965 0.601406 7.13965 0.703167V2.88815C7.13965 2.9896 7.22183 3.07178 7.32328 3.07178Z"
                fill="#7A7F86"
              />
              <path
                d="M7.32328 11.1105H13.0685C13.1696 11.1105 13.252 11.0287 13.252 10.9272V8.74223C13.252 8.64093 13.1699 8.55859 13.0685 8.55859H7.32328C7.22199 8.55859 7.13965 8.64078 7.13965 8.74223V10.9272C7.13965 11.0285 7.22183 11.1105 7.32328 11.1105Z"
                fill="#7A7F86"
              />
              <path
                d="M14.6558 12.9297H7.32328C7.22199 12.9297 7.13965 13.0114 7.13965 13.1135V15.2981C7.13965 15.3994 7.22183 15.4818 7.32328 15.4818H14.656C14.7573 15.4818 14.8394 15.3996 14.8394 15.2981V13.1132C14.8393 13.0114 14.7574 12.9297 14.6558 12.9297Z"
                fill="#7A7F86"
              />
              <path
                d="M6.08742 10.6908H4.84283V0.81072C4.84283 0.639047 4.70363 0.5 4.53211 0.5H2.02615C1.85464 0.5 1.71543 0.639047 1.71543 0.81072V10.6908H0.471C0.351218 10.6908 0.24231 10.7597 0.190731 10.8676C0.138996 10.9755 0.153755 11.1033 0.228639 11.1965L3.03692 15.3841C3.09596 15.4577 3.18498 15.5 3.27913 15.5C3.37328 15.5 3.46215 15.4577 3.52118 15.3841L6.32947 11.1965C6.40435 11.1032 6.41927 10.9755 6.36753 10.8676C6.31595 10.7598 6.2072 10.6908 6.08742 10.6908Z"
                fill="#7A7F86"
              />
            </g>
          </g>
          <defs>
            <clippath
              id="clip0_2831_2828"
            >
              <rect
                fill="white"
                height="15"
                transform="translate(0 0.5)"
                width="15"
              />
            </clippath>
          </defs>
        </svg>
        <p
          class="textValue"
        >
          По умолчанию
        </p>
      </div>
      <svg
        fill="none"
        height="16"
        viewBox="0 0 12 7"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.5 1.38477L6 5.64152L10.5 1.38477"
          stroke="#4E64F9"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
  <div
    class="btnsBlock"
  >
    <div
      class="setLimitBtns"
    >
      <button
        class="btn"
      >
        <svg
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#7A7F86"
            height="3.22302"
            rx="1"
            width="14.5036"
            y="0.5"
          />
          <rect
            fill="#7A7F86"
            height="3.22302"
            rx="1"
            width="14.5036"
            y="11.2773"
          />
          <rect
            fill="#7A7F86"
            height="3.22302"
            rx="1"
            width="14.5036"
            y="5.88867"
          />
        </svg>
        Список
      </button>
      <button
        class="btnActive"
      >
        <svg
          fill="none"
          height="14"
          viewBox="0 0 15 14"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#664EF9"
            height="5.44444"
            rx="1"
            width="5.44444"
            x="0.00390625"
          />
          <rect
            fill="#664EF9"
            height="5.44444"
            rx="1"
            width="5.44444"
            x="0.00390625"
            y="8.55469"
          />
          <rect
            fill="#664EF9"
            height="5.44444"
            rx="1"
            width="5.44444"
            x="8.55957"
          />
          <rect
            fill="#664EF9"
            height="5.44444"
            rx="1"
            width="5.44444"
            x="8.55957"
            y="8.55469"
          />
        </svg>
        Плитки
      </button>
    </div>
    <a
      class="map"
      href="/404"
    >
      <svg
        fill="none"
        height="14"
        viewBox="0 0 9 10"
        width="11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.43806 1.74419C6.7055 0.627907 5.4962 0 4.13573 0C2.7869 0 1.5776 0.627907 0.821781 1.74419C0.0659671 2.83721 -0.108452 4.23256 0.356665 5.45349C0.484572 5.77907 0.682246 6.11628 0.93806 6.4186L3.87992 9.88372C3.94969 9.95349 4.01946 10 4.12411 10C4.22876 10 4.29853 9.95349 4.36829 9.88372L7.32178 6.4186C7.5776 6.11628 7.7869 5.7907 7.90318 5.45349C8.36829 4.23256 8.19387 2.83721 7.43806 1.74419ZM4.13573 5.86047C3.13573 5.86047 2.31015 5.03488 2.31015 4.03488C2.31015 3.03488 3.13573 2.2093 4.13573 2.2093C5.13574 2.2093 5.96132 3.03488 5.96132 4.03488C5.96132 5.03488 5.14736 5.86047 4.13573 5.86047Z"
          fill="#664EF9"
        />
      </svg>
      Показать на карте
    </a>
  </div>
</section>
`;
